src/transformState.js

```diff
@@ -4,8 +4,28 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
   // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, { ...action.extraData });
+        break;
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
```

Comment:
looks better now, but `action` is object here, so why did you changed it to `for of`?
remember - use `for of` for arrays, and `for in` for objects

Reply:
![image](https://github.com/mate-academy/js_stateful-object/assets/46221837/c646a51b-9f08-4cf5-addd-0e9d29738569)
action os object but action.keysToRemove an array